docs/contributor/architecture/README.md

```diff
@@ -0,0 +1,88 @@
+
+Both set of reconcilers have to also maintain the status of the resources they reconcile. This mean that Cloud Control 
+reconcilers will observe the status of the cloud resources and project it into the status of the Cloud Control 
+resources in KCP. And that Cloud Resource reconcilers will observe the status of the Cloud Control resources and 
+project it into the status of the Cloud Resource resources in SKR.
+
+
+![API and Reconcilers](./assets/api-and-reconcilers.drawio.png "API and Reconcilers")]
+
+## KCP Cloud Control Controller Manager
+
+The Cloud Control reconcilers are managed by standard 
+[controller-runtime controller manager](https://github.com/kubernetes/sample-controller/blob/master/docs/controller-client-go.md) 
+maintaining an active long living connection to the KCP Kubernetes API.
+
+[![client-go under the hood](https://raw.githubusercontent.com/kubernetes/sample-controller/master/docs/images/client-go-controller-interaction.jpeg)](https://github.com/kubernetes/sample-controller/blob/master/docs/controller-client-go.md)
+
+# SKR Cloud Resources Controller Manager
+
+Due to the non-scalable concurrent reconciliation of large number of clusters the SKR Cloud Resources Controller Manager 
+can not keep long living connections on the remote clusters permanently watching for changes. Instead, a custom
+SKR Looper component will loop trough SKRs with enabled Cloud Resources module, and instantiate new 
+ControllerRuntime manager that will list all the "watched" (reconciler registered with manager with `.For()` or `.Watches()` 
+methods as defined in controller-runtime) and with them maintain a shot lived "cache" until all SKR reconcilers are 
+called with respective resources they are managing. Once all done, all resources for that SKR, shot living cache, client... 
+will be disposed and same process repeated for the next SKR. 
+
+The reconciler facing API like `Reconcile()` and `.SetupWithManager()` functions will remain as close as possible to 
+the one defined by controller-runtime and used by Kubebuilder. 
+
+![SKR Controller Manager](./assets/skr-controller-manager.drawio.png)
+
+
+## CloudControl Scope resource
+
+Different cloud providers APIs require different connection options to define the scope of the operations:
+* GCP - project
+* Azure - tenant and subscription
+* AWS - account
+
+The SKR loop, when projecting into the KCP resources, will set the `kymaName` field as the indication which SKR that 
+resources refers two. So, the starting point for the KCP loop reconcilers is the Kyma CR name. Following the path from 
+Kyma CR in KCP to the Shoot CR in Garden it is possible to determine the cloud provider scope. 
+
+The Cloud Control KCP resource Scope was introduced to 
+* improve performance by not reading so many different resources and reaching out to the Garden
+* simply the development landscape by not avoiding the necessity to beside local KCP cluster also have the Garden cluster 
```

Comment:
```suggestion
* simplify the development landscape by not avoiding the necessity to beside the local KCP cluster and also have the Gardener cluster 
```
I'm not sure if I follow: "by not avoiding the necessity to beside the local KCP cluster". Can we rephrase?